syft/pkg/package.go

```diff
@@ -17,18 +17,19 @@ import (
 // Package represents an application or library that has been bundled into a distributable format.
 // TODO: if we ignore FoundBy for ID generation should we merge the field to show it was found in two places?
 type Package struct {
-	id           artifact.ID      `hash:"ignore"`
-	Name         string           // the package name
-	Version      string           // the version of the package
-	FoundBy      string           `hash:"ignore" cyclonedx:"foundBy"` // the specific cataloger that discovered this package
-	Locations    file.LocationSet // the locations that lead to the discovery of this package (note: this is not necessarily the locations that make up this package)
-	Licenses     LicenseSet       // licenses discovered with the package metadata
-	Language     Language         `hash:"ignore" cyclonedx:"language"` // the language ecosystem this package belongs to (e.g. JavaScript, Python, etc)
-	Type         Type             `cyclonedx:"type"`                   // the package type (e.g. Npm, Yarn, Python, Rpm, Deb, etc)
-	CPEs         []cpe.CPE        `hash:"ignore"`                      // all possible Common Platform Enumerators (note: this is NOT included in the definition of the ID since all fields on a CPE are derived from other fields)
-	PURL         string           `hash:"ignore"`                      // the Package URL (see https://github.com/package-url/purl-spec)
-	MetadataType MetadataType     `cyclonedx:"metadataType"`           // the shape of the additional data in the "metadata" field
-	Metadata     interface{}      // additional data found while parsing the package source
+	id            artifact.ID      `hash:"ignore"`
+	Name          string           // the package name
+	Version       string           // the version of the package
+	FoundBy       string           `hash:"ignore" cyclonedx:"foundBy"` // the specific cataloger that discovered this package
+	Locations     file.LocationSet // the locations that lead to the discovery of this package (note: this is not necessarily the locations that make up this package)
+	Licenses      LicenseSet       // licenses discovered with the package metadata
+	Language      Language         `hash:"ignore" cyclonedx:"language"` // the language ecosystem this package belongs to (e.g. JavaScript, Python, etc)
+	ComponentType ComponentType    `cyclonedx:"type"`                   // the type of component (e.g. application, library, framework, etc)
```

Comment:
Hey @noqcks -- sorry for the delay here, I also left a comment on the related issue, but this gist is that instead of modifying this package, we should modify the DotnetDepsMetadata with the appropriate field(s) here: https://github.com/anchore/syft/blob/main/syft/pkg/dotnet.go#L4-L10

Reply:
are we able to access DotnetDepsMetadata from the cyclonedx formatter? from my understanding you can only access the package Struct

Reply:
Yes, you can access the metadata from the formatter, you just need to type assert the metadata field [here is an example extracting the author in CycloneDX](https://github.com/anchore/syft/blob/main/syft/format/internal/cyclonedxutil/helpers/author.go#L10-L32)